### ISSUE TITLE
Since NS7 update, some plugins fail building with "Error: spawn ./gradlew EACCES"

### ISSUE THREAD

**Issue by SzaboKrisztian** [NativeScript/NativeScript#8814]
I personally haven't tested it on Windows, so there might not be any problems on that platform because the build scripts are handled separately.

Since the update to NS7 a few days ago, there have been multiple people on the NSCommunity Slack reporting this. I personally have encountered the problem in relation to `nativescript-socket.io`, but it's not specific to it, as it actually boils down to an issue in the CLI.

The problem is two-fold: first, the EACCES error is caused by the `gradlew` file not being being marked as executable. Once this is fixed, errors relating to the carriage return character start popping up. The issue now is that the `gradlew` file uses windows style newline (`\r\n`) instead of *nix style newline (`\n`). I've implemented a quick and dirty fix, but this only solves the symptoms, it doesn't solve the problem at the root cause as I'm not familiar enough with NS to know which part of the tooling generates the `gradlew` file when building plugins.

The temporary fix involves editing a file in the NS CLI: 
1. in the global node_modules folder (unless NS is locally installed under the project's devDependencies), find `nativescript/lib/services/android-plugin-build-service.js`
2. anywhere at the top of the file (maybe between lines 19-24 for consistency's sake) add `const fs = require("fs");`
3. at line 413 (formerly 412 before the above addition) add:
```javascript
if (!this.$hostInfo.isWindows) {
    const gradleFile = path.join(pluginBuildSettings.pluginDir, gradlew);
    let content = fs.readFileSync(gradleFile, {encoding:'utf8', flag:'r'});
    content = content.replace(/\r/g, '');
    fs.writeFileSync(gradleFile, content);
    fs.chmodSync(gradleFile, 0o755, (err) => {
        if (err) {
            console.log("Error changing setting gradlew as executable");
            throw err;
        }
    });
}
```

**Comment by madmas** [comment]
Thank for writing this don, just helped me to get around this issue.

The gradle files come from here: https://github.com/NativeScript/nativescript-cli/tree/master/vendor/gradle-plugin - but that hasn't changed in a while. Except if the release was not done from there (as there is no tag nor a release in that repo).

**Comment by SzaboKrisztian** [comment]
You are right, that file was last changed over 3 years. Then my hunch would be that git + windows is responsible for this, somewhere in the NS development pipeline, but it's really hard to figure out where exactly. I'm saying this because git changes newline characters according to the platform you pull on. This might also explain why `gradlew` is stripped of its executable flag.

**Comment by SzaboKrisztian** [comment]
To make fixing this easier (also for my own sake, if I update / reinstall the NS's CLI tools), save the following as `~/ns.patch`:
```
--- ./android-plugin-build-service.js	2020-09-04 14:37:53.934958415 +0200
+++ ./fix/android-plugin-build-service.js	2020-09-04 14:36:20.008290122 +0200
@@ -17,6 +17,7 @@
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.AndroidPluginBuildService = void 0;
 const path = require("path");
+const fs = require("fs");
 const constants_1 = require("../constants");
 const helpers_1 = require("../common/helpers");
 const xml2js_1 = require("xml2js");
@@ -409,6 +410,18 @@
             if (this.$logger.getLevel() === "INFO") {
                 localArgs.push("--quiet");
             }
+            if (!this.$hostInfo.isWindows) {
+                const gradleFile = path.join(pluginBuildSettings.pluginDir, gradlew);
+                let content = fs.readFileSync(gradleFile, {encoding:'utf8', flag:'r'});
+                content = content.replace(/\r/g, '');
+                fs.writeFileSync(gradleFile, content);
+                fs.chmodSync(gradleFile, 0o755, (err) => {
+                    if (err) {
+                    console.log("Error changing setting gradlew as executable");
+                    throw err;
+                    }
+                });
+            }
             try {
                 yield this.$childProcess.spawnFromEvent(gradlew, localArgs, "close", {
                     cwd: pluginBuildSettings.pluginDir,
```

Then `cd` into where NS's CLI is installed (probably your global `node_modules` folder). For example mine is found under `/home/chris/.nvm/versions/node/v12.18.2/lib/node_modules/nativescript/` as I use nvm. Finally run the following command:
```patch -u -b lib/services/android-plugin-build-service.js -i ~/ns.patch```
